scipy/special/orthogonal.py

```diff
@@ -525,12 +540,454 @@ def h_roots(n, mu=False):
+        return _gen_roots_and_weights(m, mu0, an_func, bn_func, f, df, True, mu)
+    else:
+        nodes, weights = h_roots_asy(n)
+        if mu:
+            return nodes, weights, sum(weights)
+        else:
+            return nodes, weights
+
+
+def compute_tauk(n, k, maxit=5):
+    """Helper function for Tricomi initial guesses
+
+    For details, see formula 3.1 in lemma 3.1 in the
+    original paper.
+
+    Parameters
+    ----------
+    n : int
+        Quadrature order
+    k : ndarray of type int
+        Index of roots :math:`\tau_k` to compute
+    maxit : int
+        Number of Newton maxit performed, the default
+        value of 5 is sufficient.
+
+    Returns
+    -------
+    tauk : ndarray
+        Roots of equation 3.1
+
+    See Also
+    --------
+    initial_nodes_a
+    h_roots_asy
+    """
+    a = n % 2 - 0.5
+    c = (4.0*floor(n/2.0) - 4.0*k + 3.0)*pi / (4.0*floor(n/2.0) + 2.0*a + 2.0)
+    f = lambda x: x - sin(x) - c
+    df = lambda x: 1.0 - cos(x)
+    xi = 0.5*pi
+    for i in range(maxit):
+        xi = xi - f(xi)/df(xi)
+    return xi
+
+
+def initial_nodes_a(n, k):
+    """Tricomi initial guesses
+
+    Computes an initial approximation to the square of the `k`-th
+    (positive) root :math:`x_k` of the Hermite polynomial :math:`H_n`
+    of order :math:`n`. The formula is the one from lemma 3.1 in the
+    original paper. The guesses are accurate except in the region
+    near :math:`\sqrt{2n + 1}`.
+
+    Parameters
+    ----------
+    n : int
+        Quadrature order
+    k : ndarray of type int
+        Index of roots to compute
+
+    Returns
+    -------
+    xksq : ndarray
+        Square of the approximate roots
+
+    See Also
+    --------
+    initial_nodes
+    h_roots_asy
+    """
+    tauk = compute_tauk(n, k)
+    sigk = cos(0.5*tauk)**2
+    a = n % 2 - 0.5
+    nu = 4.0*floor(n/2.0) + 2.0*a + 2.0
+    # Initial approximation of Hermite roots (square)
+    xksq = nu*sigk - 1.0/(3.0*nu) * (5.0/(4.0*(1.0-sigk)**2) - 1.0/(1.0-sigk) - 0.25)
+    return xksq
+
+
+def compute_am(m):
+    """Approximation of the roots of the Airy function
+
+    Computes an asymptotic approximation to the root :math:`a_m`
+    of the Airy function :math:`Ai(x)` for large values of :math:`m`.
+    This formula is essentially exact for :math:`m > 10`.
+
+    Parameters
+    ----------
+    m : ndarray of type int
+        Index of the root :math:`a_m`
+
+    Returns
+    -------
+    am : ndarray
+        The `m`-th root :math:`a_m` of the function :math:`Ai(x)`
+
+    See Also
+    --------
+    airy_root
+    h_roots_asy
+    """
+    sm = 3*pi*(4*m-1) / 8.0
+    coeffs = row_stack([1.0, 5.0/48.0, -5.0/36.0, 77125.0/82944.0,
+                        -108056875.0/6967296.0, 162375596875.0/334430208.0])
+    smp = column_stack([ones_like(sm), sm**(-2), sm**(-4), sm**(-6), sm**(-8), sm**(-10)])
+    am = -sm**(2.0/3.0) * dot(smp, coeffs).reshape((-1,))
+    return am
+
+
+def airy_root(m):
```

Comment:
`special.ai_zero` is about 10x slower than this, presumably because it does not just apply the asymptotic expansion.
However, it would probably be fairly easy to speed that up.